pyemma/coordinates/transform/nystroem_tica.py

```diff
@@ -0,0 +1,921 @@
+            used in the Nystroem approximation, but may be smaller if truncate=True.
+
+        """
+        L = self.approximate_cholesky(epsilon=epsilon)
+        LL = np.dot(L.T, L)
+        s, V = np.linalg.eigh(LL)
+        # sort
+        s, V = sort_by_norm(s, V)
+
+        # back-transform eigenvectors
+        Linv = np.linalg.pinv(L.T)
+        V = np.dot(Linv, V)
+
+        # normalize eigenvectors
+        ncol = V.shape[1]
+        for i in range(ncol):
+            if not np.allclose(V[:, i], 0):
+                V[:, i] /= np.sqrt(np.dot(V[:, i], V[:, i]))
+
+        return s, V
+
+
+class SelectionStrategy:
+    def __init__(self, oasis_obj, strategy='spectral-oasis', nsel=1, neig=None):
+        """ Abstract selection strategy class
+
+        Parameters
+        ----------
+        oasis_obj : oASIS_Nystroem
+            The associated oASIS_Nystroem object.
+        strategy : str, optional, default 'spectral-oasis'
+            One of the following strategies to select new columns:
+            random : randomly choose from non-selected columns
+            oasis : maximal approximation error in the diagonal of :math:`A`
+            spectral-oasis : selects the nsel columns that are most distance in the oASIS-error-scaled dominant eigenspace
+        nsel : int, optional, default 1
+            Number of columns to be selected in each round
+        neig : int or None, optional, default None
+            Number of eigenvalues to be optimized by the selection process. Only used in 'spectral-oasis'.
+            If None, use the whole available eigenspace
+
+        """
+        self._oasis_obj = oasis_obj
+        self._strategy = strategy
+        self._nsel = nsel
+        self._neig = neig
+
+    @property
+    def strategy(self):
+        return self._strategy
+
+    @property
+    def nsel(self):
+        return self._nsel
+
+    def _check_nsel(self):
+        if self._nsel > self._oasis_obj._n - self._oasis_obj._k:
+            # less columns left than requested
+            ncols = self._oasis_obj._n - self._oasis_obj._k
+            warnings.warn('Requested more columns than are left to select. Returning only '+str(ncols)+' columns.')
+            return ncols
+        # nothing left to select?
+        if self._oasis_obj._n == self._oasis_obj._k:
+            warnings.warn('Requested more columns but there are none left. Returning None.')
+            return None
+        return self._nsel
+
+    def select(self):
+        """ Selects next column indexes according to defined strategy
+
+        Returns
+        -------
+        cols : ndarray((nsel,), dtype=int)
+            selected columns
+
+        """
+        err = self._oasis_obj.error
+        # check nsel
+        nsel = self._check_nsel()
+        if nsel is None:
+            return None
+        # go
+        return self._select(nsel, err)
+
+    def _select(self, nsel, err):
+        """ Override me to do selection """
+        pass
```

Comment:
raise NotImplementedError here, so this fails, if a deriving class forgot to overload this.